modules/tensor_mechanics/src/userobjects/GeneralizedPlaneStrainUO.C

```diff
@@ -0,0 +1,63 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "GeneralizedPlaneStrainUO.h"
+
+// libmesh includes
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<GeneralizedPlaneStrainUO>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription("Generalized Plane Strain UserObject to provide Residual and diagonal Jacobian entry");
+  params.addParam<FunctionName>("function", "0", "Function used to prescribe pressure in the out-of-plane direction");
+  params.addParam<Real>("factor", 1.0, "Scale factor applied to prescribed pressure");
```

Comment:
I'd prefer to have this named `pressure` or something that describes what the _purpose_ is rather than what kind of object is expected (we already have that information in the parameter type).